### ISSUE TITLE
Sprite Rendering Imprecision(?) (especially on bound Helpers)

### ISSUE THREAD

**Issue by GodzillaFightn** [ikemen-engine/Ikemen-GO#1751]
### Describe the bug

Small, usually imperceptible rounding(?) error during sprite rendering in non-integer positions.
Generally unimportant, but is more easily noticed and can potentially look hideous on small-scale and low-detail sprites.

### To Reproduce

1. Download purpose-made character renderBug ( https://www.mediafire.com/file/i5jhaknnlfv408n/renderBug.rar/file )
2. Play as it in Training mode
3. Jump, and go frame by frame at the peak. Pink pixels are normally hidden, but become visible during rendering errors. Slight dependency on resolution but should always be visible to some degree

### Expected behavior

No rendering imprecision, as this only started happening at all during 0.99RC builds after 0.98.2.

### Screenshots / Video

Screenshot from fullgame ft. bug to show why I would bother making an issue about a minor rendering error, the left being how it normally appears and the right being the shader used in-game:
![ikemen render bug](https://github.com/ikemen-engine/Ikemen-GO/assets/25160935/3cf64515-d839-4ea3-818a-430a3ff3b6eb)

### Engine Version (or source code date)

0.99RC builds, 0.99 itself, and current Nightly build as of issue creation

### Operating system

Windows

### Extra context or search terms

This is especially noticeable on Helpers that bind to a parent/root, despite the fact they should be in the exact same position.
Currently preventing me from upgrading past 0.98.2, as it would require compromising the visual style to an unacceptable degree and/or rewriting pos/vel with vars to ensure it only ever uses integers.

**Comment by potsmugen** [comment]
I think this just comes with the territory when we have support for characters in any resolution. Maybe separating video and game resolution would address this.

**Comment by GodzillaFightn** [comment]
Fair enough. I went and checked on a 1:1 scale resolution, and it seems to happen there too unfortunately... I thought I remembered it working there at least, but alas.
( hard to do outside of a fullgame so for reference, another screenshot )
![Screenshot_12](https://github.com/ikemen-engine/Ikemen-GO/assets/25160935/c8105dd0-0464-4830-ab4e-7f0016e331b9)

**Comment by potsmugen** [comment]
The test case looks fine after https://github.com/ikemen-engine/Ikemen-GO/pull/3601 and related changes.
